tensorflow/python/data/experimental/ops/optimization_options.py

```diff
@@ -29,19 +30,19 @@ class _AutotuneAlgorithm(enum.Enum):
   HILL_CLIMB = 0
   GRADIENT_DESCENT = 1
 
-
-@tf_export("data.experimental.OptimizationOptions")
+@deprecation.deprecated_endpoints("data.experimental.OptimizationOptions")
```

Comment:
Now that the OptimizationOptions will not be an experimental attribute of Options, some of the OptimizationOptions attribute should have an experimental prefix (so that they are not subject to backwards compatibility).

This includes:
- `filter_fusion`
- `filter_with_random_uniform_fusion`
- `hoist_random_uniform`
- `map_and_filter_fusion`
- `map_fusion`
- `reorder_data_discarding_ops`
- `parallel_batch` (which should be renamed to `experimental_batch_copy_parallelization`)

Furthermore, the autotuning options should be moved to a separate `AutotuneOptions` options object, which contains the following attributes:
- `enabled`
- `experimental_autotune_buffers`
- `cpu_budget`
- `memory_budget`

and is added to the top-level Options object as `autotune` attribute.